node/pkg/driver/node_utils.go

```diff
@@ -166,3 +169,50 @@ func (n NodeUtils) StageInfoFileIsExist(filePath string) bool {
 	}
 	return true
 }
+
+func (n NodeUtils) ParseFCPortsName(path string) ([]string, error) {
+	//FC port path is /sys/class/fc_host/host*/port_name
+	var errStrings []string
+	var FCPorts []string
+
+	fpaths, err := n.Executer.FilepathGlob(path)
+	if fpaths == nil {
+		err = fmt.Errorf(ErrorUnsupportedConnectivityType, "FC")
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	for _, fpath := range fpaths {
+		file, err := os.Open(fpath)
+		if err != nil {
+			errStrings = append(errStrings, err.Error())
+			break
+		}
+		defer file.Close()
+
+		file_out, err := ioutil.ReadAll(file)
+		if err != nil {
+			errStrings = append(errStrings, err.Error())
+			break
+		}
+
+		fileSplit := strings.Split(string(file_out), "0x")
+		if len(fileSplit) != 2 {
+			err := fmt.Errorf(ErrorWhileTryingToReadFC, string(file_out))
+			errStrings = append(errStrings, err.Error())
+		} else {
+			FCPorts = append(FCPorts, strings.TrimSpace(fileSplit[1]))
```

Comment:
if its fc port on the host, then ignore offline port could be problematic since its a transit status that can be change due to SAN temporary issue. So i think its ok to put all fc ports even if they are not online.
thoughts?

Reply:
> if its fc port on the host, then ignore offline port could be problematic since its a transit status that can be change due to SAN temporary issue. So i think its ok to put all fc ports even if they are not online.
> thoughts?

Agree with you partially. But returning an offline port will also lead to problems getting hosts from storage side according to the given ports during controllerPublish.
Actually, getting host by ports is not a good idea for FC, since FC have many ports and may have different states, defining hosts are choosing ports is not an obvious thing.